src/main/java/org/assertj/core/api/AbstractFileAssert.java

```diff
@@ -1328,6 +1328,27 @@ public SELF isNotEmpty() {
     return myself;
   }
 
+  /**
+   * Returns ByteArray assertions on the content of the actual {@code File} read.
+   * <p>
+   * Example:
+   * <pre><code class='java'> File xFile = Files.write(Paths.get("xfile.txt"), "The Truth Is Out There".getBytes()).toFile();
+   *
+   * // assertion succeeds
+   * assertThat(xFile).binaryContent().isEqualTo("The Truth Is Out There".getBytes());
+   *
+   * // assertion fails:
+   * assertThat(xFile).binaryContent().isEqualTo("Elsewhere".getBytes());</code></pre>
+   *
+   * @return a ByteArrayAssert object with the binary content of the file.
+   * @throws AssertionError if the actual {@code File} is not readable as per {@link java.nio.file.Files#isReadable(java.nio.file.Path)}.
+   * @throws UncheckedIOException when failing to read the actual {@code File}.
+   */
+  public AbstractByteArrayAssert<?> binaryContent() {
+    files.assertCanRead(info, actual);
+    return new ByteArrayAssert(readFile()).withAssertionState(myself);
```

Comment:
we need to test soft assertions as this is a navigation method, add a test method file_soft_assertions_should_work_with_binaryContent` like `SoftAssertionsTest.file_soft_assertions_should_work_with_navigation_methods`

the test should fail because `binaryContent` is not registered in `SoftProxies.METHODS_CHANGING_THE_OBJECT_UNDER_TEST`, test should pass after you add it.

In the same way, add `Assumptions_assumeThat_with_File_binaryContent_Test` similar to `Assumptions_assumeThat_with_File_content_Test`

Reply:
Should be fixed in Should be fixed in https://github.com/assertj/assertj-core/pull/2558/commits/ca26945da0ab3ff57d5f1c0de48045b425b0156e